packages/scene-composer/src/store/Store.ts

```diff
@@ -62,24 +63,33 @@ export type RootState = ISharedState &
  * Core state management functions
  * TODO: make them into slices and better organized
  */
-const stateCreator: StateCreator<RootState> = (set, get, api) => ({
-  lastOperation: undefined,
-  ...createSceneDocumentSlice(set, get),
-  ...createEditStateSlice(set, get, api),
-  ...createDataStoreSlice(set, get, api),
-  noHistoryStates: {
-    ...createViewOptionStateSlice(set),
-  },
-  ...createNodeErrorStateSlice(set, get, api),
-});
 
-const createStateImpl: () => UseStore<RootState> = () => create<RootState>(undoMiddleware(log(immer(stateCreator))));
+type StoreState<T> = {
+  (): T;
+  <U>(selector: (state: T) => U): U;
+  <U>(selector: (state: T) => U, equalityFn: (a: U, b: U) => boolean): U;
+} & StoreApi<T>;
+
+const createStateImpl: () => StoreState<RootState> = () =>
+  create<RootState>()(
+    undoMiddleware(
+      log(
+        immer((...args) => ({
+          ...createViewOptionStateSlice(...args),
+          ...createSceneDocumentSlice(...args),
+          ...createEditStateSlice(...args),
+          ...createDataStoreSlice(...args),
+          ...createNodeErrorStateSlice(...args),
+        })) as StateCreator<RootState>,
+      ),
+    ),
+  );
 
 // TODO: currently undoMiddleware will record editor state changes, such as select/deselect object.
 // We may want to fine-tune the undo/redo experience.
-const stores = new Map<string, UseStore<RootState>>();
+const stores = new Map<string, StoreState<RootState>>();
 
-const useStore: (id: string) => UseStore<RootState> = (id: string) => {
+const useStore = (id: string) => {
```

Comment:
Can we not set the type on useStore anymore?  It's not something like StoreState<RootState> instead now?